packages/catalog-client/src/CatalogClients.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, EntityName } from '@backstage/catalog-model';
+import {
+  CatalogClient,
+  EntitiesRequestOption,
+  EntitiesResponse,
+} from './types';
+import fetch from 'cross-fetch';
+import { Config } from '@backstage/config';
+import { EntityFilters } from './EntityFilters';
+
+/**
+ * A client to communicate with the catalog.
+ */
+export class CatalogClients implements CatalogClient {
```

Comment:
Naming things when you want the interface and the implementation to be the same, bleh.

Could make the interface be `CatalogClientApi` so we can name this `CatalogClient`?

Another option would be to make this class only contain the static method, and then have a separate `CatalogHttpClient` or smth that's just an internal implementation that the factory returns

Reply:
Yuup the naming is my main gripe with the "interface rich" apis.

First option sounds best I think. I'll experiment. Not sure it's worth breaking the impl from the static factory function

Reply:
`ICatalogClient` :trollface:

Wouldn't mind coming up with a convention though, and also likely wouldn't mind that convention being an `*Api` suffix :grin:

Reply:
Orrrr `CatalogClientT` 🥇

Reply:
Haha

Reply:
Quite a few plugins do this and I've commonly seen the implementation use a suffix like RestApi or just Client. My vote would be for using an Api suffix for the interface and maybe Client for the implementation. The term API seems to jive with how they are modelled as entities as well imo.

Reply:
You'd use the `catalogClientApiRef` to get a `CatalogClientApi` type (actually implemented by a `CatalogClient` class)?

Reply:
Hmm 🤔. Maybe `catalogApiRef` to get a `CatalogApi` with an implementation of `CatalogApiClient`??